src/router.js

```diff
@@ -1,57 +1,106 @@
 import express from 'express';
-import { checkSchema } from 'express-validator';
+import { checkSchema, validationResult } from 'express-validator';
+import log from 'loglevel';
 import db from './db';
-import postSchema from './validation/schema_postMotivator';
+import expectedMotivatorFormat from './validation/schema_postMotivator';
+import expectedMotivatorResultFormat from './validation/schema_postMotivatorResult';
+import motivatorResultsForUser from './dbHelper';
+import NotFoundError from './errors/NotFoundError';
 
 const router = express.Router();
 
-router.get('/', (req, res) => {
-  db.models.motivator.findAll({
-    include: [
-      {
-        model: db.models.motivatorContent,
-      },
-    ],
-  }).then((query) => {
-    const response = query.map((item) => item.toJSON())
-      .map((item) => {
-        const content = item.MotivatorContents.map((x) => JSON.parse(x.content));
-
-        /*
-
-        const result = item.MotivatorResults.map(({
-          motivator_id, user_id, MotivatorResultInputs, ...resultItem
-        }) => ({ ...resultItem }));
-
-        const inputs = item.MotivatorResults[0].MotivatorResultInputs
-          .map(({ value }) => ({ ...JSON.parse(value) }));
-
-        return {
-          ...item, content, result: result[0], inputs,
-        };
-
-        */
-        return { ...item, content };
-      })
-      .map(({ MotivatorContents, ...itemWithoutContents }) => ({
-        ...itemWithoutContents,
-      }));
-    res.send(response);
-  }).catch((error) => {
-    res.status(500).send(error);
+router.get('/', async (req, res) => {
+  const query = await db.models.motivator.findAll({
+    include: [{ model: db.models.motivatorContent }, {
+      model: db.models.motivatorResult,
+      where: { user_id: req.user.uid },
+      limit: 1,
+      include: [{ model: db.models.motivatorResultInput }],
+    }],
   });
+
+  const response = query.map(((item) => item.toJSON()))
+    .map((item) => ({
+      ...item,
+      content: item.MotivatorContents.map((x) => JSON.parse(x.content)),
+      result: item.MotivatorResults.map(({
+        motivator_id, user_id, MotivatorResultInputs, ...resItem
+      }) => ({ ...resItem })).first(),
+      inputs: item.MotivatorResults[0].MotivatorResultInputs.map(({ value }) => ({
+        ...JSON.parse(value),
+      })),
+    }))
+    .map(({ MotivatorContents, MotivatorResults, ...itemWithoutContents }) => ({
+      ...itemWithoutContents,
+    }));
```

Comment:
```suggestion
    }))
    .map(({ MotivatorContents, ...itemWithoutContents }) => ({
      ...itemWithoutContents,
    }));
```

see above. returning result data here is not desired. there is a reason I commented it out :)